include/deal.II/grid/manifold_lib.h

```diff
@@ -600,6 +600,15 @@ class FunctionManifold : public ChartManifold<dim, spacedim, chartdim>
     const Tensor<1, chartdim> &periodicity = Tensor<1, chartdim>(),
     const double               tolerance   = 1e-10);
 
+  /**
+   * Same as previous, but takes pointers to the functions.
+   */
+  FunctionManifold(
+    std::unique_ptr<Function<chartdim>> &push_forward_function,
+    std::unique_ptr<Function<spacedim>> &pull_back_function,
```

Comment:
I think this makes sense. But it would be useful to add something along the lines of
```
  This function takes ownership of the Function objects passed as first and second argument,
  and is ultimately in charge of deleting the pointers when the FunctionManifold object is
  destroyed. 
```
Maybe also something like
```
  This constructor is useful because it allows creating function objects at the place
  of calling the constructor without having to name and later delete these objects.
  This allows the following idiom:
     FunctionManifold<dim> manifold (std_cxx14::make_unique<MyPushForward>(...),
                                                           std_cxx14::make_unique<MyPullBack>(...));
```